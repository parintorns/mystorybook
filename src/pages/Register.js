import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  StyleSheet,
  Image,
  Alert
} from 'react-native'
import {
  Container,
  Content,
  Header,
  Left,
  Body,
  Title,
  Right,
  View,
  Button,
  Icon,
  Text,
  Form,
  Item,
  Input
} from 'native-base'

export class Register extends Component {
  constructor() {
    super()

    this.handleGoBack = this.handleGoBack.bind(this)
    this.handleGoNext = this.handleGoNext.bind(this)
  }

  handleGoBack () {
    const self = this

    Alert.alert(
      'Cancel Registration',
      'Your input data will clear and go back to login page. Are you sure?',
      [
        {text: 'Cancel'},
        {text: 'OK', onPress: () => self.props.navigation.navigate('Login')}
      ],
      { cancelable: false }
    )
  }

  handleGoNext () {
    this.props.navigation.navigate('Feed')
  }

  render () {
    return (
      <Container>
        <Header>
          <Left>
            <Button transparent
              id='btnBack'
              onPress={this.handleGoBack}>
              <Title>Cancel</Title>
            </Button>
          </Left>
          <Right>
            <Button transparent
              id='btnNext'
              onPress={this.handleGoNext}>
              <Title>Next</Title>
              <Icon
                name='ios-arrow-round-forward-outline'
              />
            </Button>
          </Right>
        </Header>
        <Content padder>
          <View style={styles.avatarWrapper}>
            <Image style={styles.avatar} />
          </View>
          <Form>
            <Item first>
              <Input placeholder="Firstname" />
            </Item>
            <Item>
              <Input placeholder="Lastname" />
            </Item>
            <Item>
              <Input placeholder="Phone" />
            </Item>
            <Item>
              <Input placeholder="City" />
            </Item>
            <Item last>
              <Input placeholder="Country" />
            </Item>
          </Form>
        </Content>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  avatarWrapper: {
    alignItems: 'center',
    marginVertical: 20
  },
  avatar: {
    height: 140,
    width: 140,
    aspectRatio: 1,
    borderRadius: 70,
    backgroundColor: '#aaa'
  }
})


Register.propTypes = {
  navigation: PropTypes.object.isRequired
}

export default Register
