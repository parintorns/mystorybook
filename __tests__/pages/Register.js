import React from 'react'
import { shallow } from 'enzyme'

import Register from '../../src/pages/Register'

const spyHandleGoBack = jest.spyOn(Register.prototype, 'handleGoBack')
const spyHandleGoNext = jest.spyOn(Register.prototype, 'handleGoNext')

const props = {
  navigation: {
    navigate: jest.fn()
  }
}

let wrapper

beforeEach(() => {
  wrapper = shallow(
    <Register
      {...props}
    />
  )
})

describe('Render', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Actions', () => {
  it('should go to back' ,() => {
    const btnBack = wrapper.find({ id: 'btnBack'})
    expect(btnBack).toHaveLength(1)

    btnBack.simulate('press')
    expect(spyHandleGoBack).toHaveBeenCalledTimes(1)
  })

  it('should go to next', () => {
    const btnNext = wrapper.find({ id: 'btnNext'})
    expect(btnNext).toHaveLength(1)

    btnNext.simulate('press')
    expect(spyHandleGoNext).toHaveBeenCalledTimes(1)
  })
})
