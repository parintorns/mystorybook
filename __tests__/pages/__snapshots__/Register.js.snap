// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    navigation={
      Object {
        "navigate": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Header)
          virtual={undefined}
        >
          <Styled(Left)
            virtual={undefined}
          >
            <Styled(Button)
              id="btnBack"
              onPress={[Function]}
              transparent={true}
              virtual={undefined}
            >
              <Styled(Title)
                virtual={undefined}
              >
                Cancel
              </Styled(Title)>
            </Styled(Button)>
          </Styled(Left)>
          <Styled(Right)
            virtual={undefined}
          >
            <Styled(Button)
              id="btnNext"
              onPress={[Function]}
              transparent={true}
              virtual={undefined}
            >
              <Styled(Title)
                virtual={undefined}
              >
                Next
              </Styled(Title)>
              <Styled(Icon)
                name="ios-arrow-round-forward-outline"
                virtual={undefined}
              />
            </Styled(Button)>
          </Styled(Right)>
        </Styled(Header)>,
        <Styled(Content)
          padder={true}
          virtual={undefined}
        >
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "center",
                "marginVertical": 20,
              }
            }
            virtual={undefined}
          >
            <Image
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#aaa",
                  "borderRadius": 70,
                  "height": 140,
                  "width": 140,
                }
              }
            />
          </Styled(ViewNB)>
          <Styled(Form)
            virtual={undefined}
          >
            <Styled(Item)
              first={true}
              virtual={undefined}
            >
              <Styled(Input)
                placeholder="Firstname"
                virtual={undefined}
              />
            </Styled(Item)>
            <Styled(Item)
              virtual={undefined}
            >
              <Styled(Input)
                placeholder="Lastname"
                virtual={undefined}
              />
            </Styled(Item)>
            <Styled(Item)
              virtual={undefined}
            >
              <Styled(Input)
                placeholder="Phone"
                virtual={undefined}
              />
            </Styled(Item)>
            <Styled(Item)
              virtual={undefined}
            >
              <Styled(Input)
                placeholder="City"
                virtual={undefined}
              />
            </Styled(Item)>
            <Styled(Item)
              last={true}
              virtual={undefined}
            >
              <Styled(Input)
                placeholder="Country"
                virtual={undefined}
              />
            </Styled(Item)>
          </Styled(Form)>
        </Styled(Content)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Left)
              virtual={undefined}
            >
              <Styled(Button)
                id="btnBack"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Cancel
                </Styled(Title)>
              </Styled(Button)>
            </Styled(Left)>,
            <Styled(Right)
              virtual={undefined}
            >
              <Styled(Button)
                id="btnNext"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Next
                </Styled(Title)>
                <Styled(Icon)
                  name="ios-arrow-round-forward-outline"
                  virtual={undefined}
                />
              </Styled(Button)>
            </Styled(Right)>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                id="btnBack"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Cancel
                </Styled(Title)>
              </Styled(Button)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Title)
                  virtual={undefined}
                >
                  Cancel
                </Styled(Title)>,
                "id": "btnBack",
                "onPress": [Function],
                "transparent": true,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Cancel",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Cancel",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                id="btnNext"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Next
                </Styled(Title)>
                <Styled(Icon)
                  name="ios-arrow-round-forward-outline"
                  virtual={undefined}
                />
              </Styled(Button)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Title)
                    virtual={undefined}
                  >
                    Next
                  </Styled(Title)>,
                  <Styled(Icon)
                    name="ios-arrow-round-forward-outline"
                    virtual={undefined}
                  />,
                ],
                "id": "btnNext",
                "onPress": [Function],
                "transparent": true,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Next",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Next",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "ios-arrow-round-forward-outline",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(ViewNB)
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 20,
                }
              }
              virtual={undefined}
            >
              <Image
                style={
                  Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#aaa",
                    "borderRadius": 70,
                    "height": 140,
                    "width": 140,
                  }
                }
              />
            </Styled(ViewNB)>,
            <Styled(Form)
              virtual={undefined}
            >
              <Styled(Item)
                first={true}
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Firstname"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Lastname"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Phone"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="City"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                last={true}
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Country"
                  virtual={undefined}
                />
              </Styled(Item)>
            </Styled(Form)>,
          ],
          "padder": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                style={
                  Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#aaa",
                    "borderRadius": 70,
                    "height": 140,
                    "width": 140,
                  }
                }
              />,
              "style": Object {
                "alignItems": "center",
                "marginVertical": 20,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#aaa",
                  "borderRadius": 70,
                  "height": 140,
                  "width": 140,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Item)
                  first={true}
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Firstname"
                    virtual={undefined}
                  />
                </Styled(Item)>,
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Lastname"
                    virtual={undefined}
                  />
                </Styled(Item)>,
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Phone"
                    virtual={undefined}
                  />
                </Styled(Item)>,
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="City"
                    virtual={undefined}
                  />
                </Styled(Item)>,
                <Styled(Item)
                  last={true}
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Country"
                    virtual={undefined}
                  />
                </Styled(Item)>,
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Input)
                    placeholder="Firstname"
                    virtual={undefined}
                  />,
                  "first": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "placeholder": "Firstname",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Input)
                    placeholder="Lastname"
                    virtual={undefined}
                  />,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "placeholder": "Lastname",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Input)
                    placeholder="Phone"
                    virtual={undefined}
                  />,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "placeholder": "Phone",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Input)
                    placeholder="City"
                    virtual={undefined}
                  />,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "placeholder": "City",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Input)
                    placeholder="Country"
                    virtual={undefined}
                  />,
                  "last": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "placeholder": "Country",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Header)
            virtual={undefined}
          >
            <Styled(Left)
              virtual={undefined}
            >
              <Styled(Button)
                id="btnBack"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Cancel
                </Styled(Title)>
              </Styled(Button)>
            </Styled(Left)>
            <Styled(Right)
              virtual={undefined}
            >
              <Styled(Button)
                id="btnNext"
                onPress={[Function]}
                transparent={true}
                virtual={undefined}
              >
                <Styled(Title)
                  virtual={undefined}
                >
                  Next
                </Styled(Title)>
                <Styled(Icon)
                  name="ios-arrow-round-forward-outline"
                  virtual={undefined}
                />
              </Styled(Button)>
            </Styled(Right)>
          </Styled(Header)>,
          <Styled(Content)
            padder={true}
            virtual={undefined}
          >
            <Styled(ViewNB)
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 20,
                }
              }
              virtual={undefined}
            >
              <Image
                style={
                  Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#aaa",
                    "borderRadius": 70,
                    "height": 140,
                    "width": 140,
                  }
                }
              />
            </Styled(ViewNB)>
            <Styled(Form)
              virtual={undefined}
            >
              <Styled(Item)
                first={true}
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Firstname"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Lastname"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Phone"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="City"
                  virtual={undefined}
                />
              </Styled(Item)>
              <Styled(Item)
                last={true}
                virtual={undefined}
              >
                <Styled(Input)
                  placeholder="Country"
                  virtual={undefined}
                />
              </Styled(Item)>
            </Styled(Form)>
          </Styled(Content)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Left)
                virtual={undefined}
              >
                <Styled(Button)
                  id="btnBack"
                  onPress={[Function]}
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Title)
                    virtual={undefined}
                  >
                    Cancel
                  </Styled(Title)>
                </Styled(Button)>
              </Styled(Left)>,
              <Styled(Right)
                virtual={undefined}
              >
                <Styled(Button)
                  id="btnNext"
                  onPress={[Function]}
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Title)
                    virtual={undefined}
                  >
                    Next
                  </Styled(Title)>
                  <Styled(Icon)
                    name="ios-arrow-round-forward-outline"
                    virtual={undefined}
                  />
                </Styled(Button)>
              </Styled(Right)>,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  id="btnBack"
                  onPress={[Function]}
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Title)
                    virtual={undefined}
                  >
                    Cancel
                  </Styled(Title)>
                </Styled(Button)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Title)
                    virtual={undefined}
                  >
                    Cancel
                  </Styled(Title)>,
                  "id": "btnBack",
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Cancel",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  id="btnNext"
                  onPress={[Function]}
                  transparent={true}
                  virtual={undefined}
                >
                  <Styled(Title)
                    virtual={undefined}
                  >
                    Next
                  </Styled(Title)>
                  <Styled(Icon)
                    name="ios-arrow-round-forward-outline"
                    virtual={undefined}
                  />
                </Styled(Button)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Styled(Title)
                      virtual={undefined}
                    >
                      Next
                    </Styled(Title)>,
                    <Styled(Icon)
                      name="ios-arrow-round-forward-outline"
                      virtual={undefined}
                    />,
                  ],
                  "id": "btnNext",
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Next",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": "Next",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "name": "ios-arrow-round-forward-outline",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(ViewNB)
                style={
                  Object {
                    "alignItems": "center",
                    "marginVertical": 20,
                  }
                }
                virtual={undefined}
              >
                <Image
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#aaa",
                      "borderRadius": 70,
                      "height": 140,
                      "width": 140,
                    }
                  }
                />
              </Styled(ViewNB)>,
              <Styled(Form)
                virtual={undefined}
              >
                <Styled(Item)
                  first={true}
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Firstname"
                    virtual={undefined}
                  />
                </Styled(Item)>
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Lastname"
                    virtual={undefined}
                  />
                </Styled(Item)>
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Phone"
                    virtual={undefined}
                  />
                </Styled(Item)>
                <Styled(Item)
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="City"
                    virtual={undefined}
                  />
                </Styled(Item)>
                <Styled(Item)
                  last={true}
                  virtual={undefined}
                >
                  <Styled(Input)
                    placeholder="Country"
                    virtual={undefined}
                  />
                </Styled(Item)>
              </Styled(Form)>,
            ],
            "padder": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#aaa",
                      "borderRadius": 70,
                      "height": 140,
                      "width": 140,
                    }
                  }
                />,
                "style": Object {
                  "alignItems": "center",
                  "marginVertical": 20,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "style": Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#aaa",
                    "borderRadius": 70,
                    "height": 140,
                    "width": 140,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Item)
                    first={true}
                    virtual={undefined}
                  >
                    <Styled(Input)
                      placeholder="Firstname"
                      virtual={undefined}
                    />
                  </Styled(Item)>,
                  <Styled(Item)
                    virtual={undefined}
                  >
                    <Styled(Input)
                      placeholder="Lastname"
                      virtual={undefined}
                    />
                  </Styled(Item)>,
                  <Styled(Item)
                    virtual={undefined}
                  >
                    <Styled(Input)
                      placeholder="Phone"
                      virtual={undefined}
                    />
                  </Styled(Item)>,
                  <Styled(Item)
                    virtual={undefined}
                  >
                    <Styled(Input)
                      placeholder="City"
                      virtual={undefined}
                    />
                  </Styled(Item)>,
                  <Styled(Item)
                    last={true}
                    virtual={undefined}
                  >
                    <Styled(Input)
                      placeholder="Country"
                      virtual={undefined}
                    />
                  </Styled(Item)>,
                ],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Input)
                      placeholder="Firstname"
                      virtual={undefined}
                    />,
                    "first": true,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "placeholder": "Firstname",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Input)
                      placeholder="Lastname"
                      virtual={undefined}
                    />,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "placeholder": "Lastname",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Input)
                      placeholder="Phone"
                      virtual={undefined}
                    />,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "placeholder": "Phone",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Input)
                      placeholder="City"
                      virtual={undefined}
                    />,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "placeholder": "City",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Input)
                      placeholder="Country"
                      virtual={undefined}
                    />,
                    "last": true,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "placeholder": "Country",
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
